src/v2/controllers/FormController.js

```diff
@@ -248,8 +250,24 @@ export default Controller.extend({
 
     if (IonResponseHelper.isIonErrorResponse(error)) {
       const convertedErrors = IonResponseHelper.convertFormErrors(error);
-      const showBanner = convertedErrors.responseJSON.errorCauses.length ? false : true;
-      model.trigger('error', model, convertedErrors, showBanner);
+      if (IonResponseHelper.containsMessageWithI18nKey(convertedErrors, OV_UV_ENABLE_BIOMETRIC_SERVER_KEY)) {
```

Comment:
I don't think this belongs to the FormController, this seems to be specific to OktaVerify. What I can see from the last change is that we moved the code to show a callout from `showMessages` in ChallengeOktaVerifyTotpView.js to here, but I think it was fine in the OktaVerify View... 

`showMessages` is called when we re-render the view, which happens in `FormController.handleFormNameChange`, which maybe does not happen for TOTP. So what I think we can do as well is add a listener to the model in ChallengeOktaVerifyTotpView.js, something like
```
    initialize() {
      BaseForm.prototype.initialize.apply(this, arguments);
      this.listenTo(this.model, 'error', () => {
        showMessages();
      });
    },
```
Did you try that?

Reply:
yea, i agree this isn't the best place for it
i did try doing what you suggested above and it didn't work - i'm not sure if maybe at the point we are in this method we are too far removed from the view?